optimizer/optimizers.py

```diff
@@ -298,12 +304,48 @@ def _create_optimizer(self, args, local_optimizer_config, parameters):
                 amsgrad=False,
             )
 
-        log_optimizer(optimizer, betas, epsilon, weight_decay, curr_lr)
+        log_optimizer(label, optimizer, betas, epsilon, weight_decay, curr_lr)
         return optimizer
 
-def log_optimizer(optimizer: torch.optim.Optimizer, betas, epsilon, weight_decay, lr):
+    def _apply_text_encoder_freeze(self, text_encoder) -> chain[Any]:
+        parameters = itertools.chain([])
+
+        if self.te_freeze_config.get('freeze_embeddings', False):
+            # freeze embeddings
+            print(" ❄️ freezing embeddings")
```

Comment:
The emoji is cute, but will it cause problems on Colab or Vast?

Reply:
i checked Jupyter and colab and they both handle it fine